application/models/testutils.go

```diff
@@ -937,3 +938,15 @@ func Must(err error) {
 		panic(err.Error())
 	}
 }
+
+// SameAppError verifies that the actual error contains an AppError and that the key and category match expected
+func SameAppError(t *testing.T, expected api.AppError, actual error) {
```

Comment:
Could make it a bit clearer that this is an assertion / requirement

```suggestion
func AssertSameAppError(t *testing.T, expected api.AppError, actual error) {
```